File: 프로그래머스/lv0/120815. 피자 나눠 먹기 （2）/피자 나눠 먹기 （2）.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n)
{
    int pizza = 0;
    int a = n;
    if (n % 6 == 0)
        pizza = n / 6;
    else
    {
        while (1)
        {
            if (n % 6 == 0)
            {
                pizza = n / 6;
                break;
            }
            n += a;
        }
    }
    return pizza;
}

File: 프로그래머스/0/120815. 피자 나눠 먹기 （2）/피자 나눠 먹기 （2）.java
class Solution {
    public int solution(int n) {
        return lcm(n, 6) / 6;
    }
    
    private int gcd(int a, int b) {
        return b == 0 ? a : gcd(b, a % b);
    }
    
    private int lcm(int a,int b) {
        return a * b / gcd(a, b);
    } 
}